models/reporter.go

```diff
@@ -368,34 +368,36 @@ func (r *Reporter) SelectSparklines(ctx context.Context, dimensionVal string,
 }
 
 const queryDimension = `
-SELECT '{{ .DimensionName }}' AS key, {{ .DimensionName }} AS value, SUM( {{ .MainMetric }} ) AS main_metric_sum
-  FROM metrics
- WHERE period_start >= ?
-   AND period_start <= ?
-   {{range $key, $vals := .Dimensions }} AND {{ $key }} IN ( '{{ StringsJoin $vals "', '" }}' ){{ end }}
-    {{ if .Labels }}{{$i := 0}}
-       AND ({{range $key, $val := .Labels }} {{ $i = inc $i}}
-                {{ if gt $i 1}} AND {{ end }} has(['{{ StringsJoin $val "', '" }}'], labels.value[indexOf(labels.key, '{{ $key }}')])
-           {{ end }})
-    {{ end }}
-GROUP BY {{ .DimensionName }}
- WITH TOTALS
-ORDER BY main_metric_sum DESC, value;
-`
-
-const queryLabels = `
-SELECT arrayJoin(labels.key) AS key, labels.value[indexOf(labels.key, key)] AS value, SUM( {{ .MainMetric }} ) AS main_metric_sum
-  FROM metrics
- WHERE period_start >= ?
-   AND period_start <= ?
-    {{range $keyD, $valsD := .Dimensions }} AND {{ $keyD }} IN ( '{{ StringsJoin $valsD "', '" }}' ){{ end }}
-    {{ if .Labels }}{{$i := 0}} AND
-        ({{range $keyL, $valsL := .Labels }} {{ $i = inc $i}}
-            {{ if gt $i 1}} AND {{ end }} has(['{{ StringsJoin $valsL "', '" }}'], labels.value[indexOf(labels.key, '{{ $keyL }}')])
-        {{ end }})
-    {{ end }}
-GROUP BY key, value
-ORDER BY main_metric_sum DESC, key, value;
+SELECT
+    key,
+    value,
+    sum(main_metric_sum) AS main_metric_sum
+FROM
+(
+    SELECT
+        '{{ .DimensionName }}' AS key,
```

Comment:
why do we need this column? it's static, isn't it? maybe we can use `dimensionName` variable here https://github.com/percona/qan-api2/blob/3c73e89bbb0d4af5e9b9774572543d1d984e2747/models/reporter.go#L515